whois-api/src/main/java/net/ripe/db/whois/api/apiKey/BearerTokenExtractor.java

```diff
@@ -97,11 +98,21 @@ private OAuthSession getOAuthSession(final String bearerToken, final String apiK
                     apiKeyId,
                     claimSet.getStringClaim("email"),
                     claimSet.getStringClaim("uuid"),
-                    claimSet.getStringClaim("scope"));
+                    getWhoisScopes(claimSet.getStringClaim("scope")));
 
         } catch (Exception e) {
             LOGGER.info("Failed to read OAuthSession from BearerToken ", e);
             return new OAuthSession(apiKeyId);
         }
     }
+
+    private List<OAuthSession.ScopeFormatter> getWhoisScopes(final String scopes) {
```

Comment:
This will return an empty list in next both situations:
- if the scope starts by something different than whois -> NOT ALLOWED
- if the scope is empty -> ALLOWED
Is this working fine for each of the situations?
Should we directly return an error here if the scope is not empty and doesn't start by WHOIS?

Reply:
Any custom whois scope will always starts from whois and JWT already by default has scope " profile and email " which we should ignore. There are test cases for these situations. Scope can be or cannot be empty .